docs: service accounts (#2881)
Unleash/unleash: website/docs/how-to/how-to-create-service-accounts.mdx

---
title: How to create service accounts
---

:::info availability

> Service accounts is an upcoming enterprise feature, and is scheduled to become available in one of the next few releases.

Reviewer: This phrase is a bit vague, I'm not sure how we handle these but I'd do a couple of things:
1. Add a date of when this message is posted
2. If we're not sure on when it's going to be available, we can say that the release date is yet to be defined

```suggestion
**January, 2023:** Service accounts is an upcoming enterprise feature. The release date is yet to be defined.
```

Author: Thanks @gastonfournier - I think you have a point here. This was originally adapted from https://docs.getunleash.io/how-to/how-to-clone-environments - That should also probably be changed (PS: Isn't environment cloning currently GA?). 

I'll raise this discussion in Slack and tackle it on a different PR if changes are needed.

:::

[Service accounts](../reference/service-accounts.md) enable Unleash admins to create accounts that act as users and respect the same set of permissions, however they do not have a password and cannot log in to the Unleash UI. Instead, they are intended to be used to access the Unleash API programatically.

## Step 1: Navigate to the service accounts page {#step-1}

> Navigate to the _Service accounts_ page in the admin UI (available at the URL `/admin/service-accounts`). Use the "Settings Wheel" on the navigation menu and select "Service accounts".

Reviewer 2: This should probably be "settings button" instead. There's no non-visual indication that the button presents itself as a wheel, so I'm afraid it could be problematic to users with vision impairment. Plus, a settings wheel might conjure a whole different image.

```suggestion
Navigate to the _service accounts_ page in the admin UI (available at the URL `/admin/service-accounts`). Use the _settings_ button in the navigation menu and select "service accounts".
```

> ![The admin UI navigation "Settings Wheel" submenu with the Service accounts item highlighted.](/img/service-account-1.png)

Reviewer 2: ```suggestion
![The admin UI navigation settings submenu with the Service accounts item highlighted.](/img/service-account-1.png)
```

## Step 2: Click the "new service account" button {#step-2}

> On the top right corner, use the "new service account" button to open the "new service account" form.

Reviewer 2: In general, I'd suggest leaving the specific location of elements out of guides. Primarily so that we don't need to update it if we change the layout.

```suggestion
Use the "new service account" button to open the "new service account" form.
```

![The "service accounts" table with the "new service account" button highlighted.](/img/service-account-2.png)

## Step 3: Fill in the service account form {#step-3}

Give your new service account a name. After leaving the name field, the username field is pre-filled with a suggestion based on the name you entered, but you can change it to whatever you like.

Select a [global role](https://docs.getunleash.io/reference/rbac#standard-roles) for your service account, which will define what your new service account will be allowed to do. The roles that you can assign to service accounts are the same ones that are available for regular users.

> ![The service account form filled with some example data, and the Add service account button highlighted at the bottom.](/img/service-account-3.png)

Reviewer 2: ```suggestion
![The service account form filled with some example data, and the "add service account" button highlighted at the bottom.](/img/service-account-3.png)
```

You can optionally generate a token for the new service account right away. Give your token a description and optionally set an expiry date. By default the expiry date is set to 30 days. The token will be generated when you submit the form.

> ![The service account form with the Token section highlighted and the Generate a token now option selected](/img/service-account-4.png)

Reviewer 2: ```suggestion
![The service account form with the token section highlighted and the "generate a token now" option selected](/img/service-account-4.png)
```

> ![The token details with the "Copy Token" element highlighted.](/img/service-account-5.png)

Reviewer 2: ```suggestion
![The token details with the "copy token" element highlighted.](/img/service-account-5.png)
```
